### ISSUE TITLE
CAPI refitting

### ISSUE THREAD

**Issue by eightreal** [microsoft/LightGBM#6430]
Hello , I have see the LGBM_BoosterRefit api in c——API
is there more description about it? a example code is much better, I 'm  try to do continue learning by new data , but there is few doc.

**Comment by eightreal** [comment]
In python , I see some example code like 

refit(data=x_test, label=y_test)

but , how it work in C_Api, how can I get the  input
```
const int32_t* leaf_preds,
                                        int32_t nrow,
                                        int32_t ncol
                                        ```

**Comment by eightreal** [comment]
I read the python source code 
So, I need call the LGBM_BoosterPredictForMat to get the leaf index (by my new dataset )
the input lead index for refit ,
Is the workflow correct? 
and whether should I call the LGBM_BoosterResetTrainingData before call refit?

**Comment by eightreal** [comment]
OH, I also see you have call a new booster and merge.

**Comment by eightreal** [comment]
is there any member can help answer my question?

**Comment by jameslamb** [comment]
Please look through what `Booster.refit()` in the Python package does.

https://github.com/microsoft/LightGBM/blob/88cec4776e621ac93f9ba03aa0015035570545da/python-package/lightgbm/basic.py#L4747

**Comment by eightreal** [comment]
收到，谢谢

**Comment by eightreal** [comment]
I can do refit by this workflow,  but I check that you create a empty model and merge old model, so can I reset training dataset directly?
